### ISSUE TITLE
Problem with DUS step (DrugExposureDiagnosis)

### ISSUE THREAD

**Issue by JavierGorri** [oxford-pharmacoepi/MegaStudy#68]
We have run part of the code (we have done "Indications concepts" and "Indication cohort") but we have a problem with "Drug concepts". This is the problem with "DrugExposureDiagnosis".

![image](https://github.com/oxford-pharmacoepi/MegaStudy/assets/68581563/9bd58855-62bf-49c4-9908-75f46b8a47cb)

@JavierGorri

**Comment by tiozab** [comment]
@JavierGorri thank you for opening this issue. However, I do not see the error message and where the error occured from this screenshot (it is not displaying fully). 

@D229463 Are you sure the renv worked? DrugExposureDiagnostics is not part of the DUS renv.

**Comment by D229463** [comment]
@tiozab I try to explain it in more detail. We made three attempts:

1. First one with renv original libraries (as you say DrugExposureDiagnostics is not part of them) and the error was a problem with our database space. We solved it (adding more)

2. The second one with renv original libraries too and with more space in database and error was

_> source(here("drugs.R"))
Error in checkDbType(cdm = cdm, type = "cdm_reference", messageStore = errorMessage) : 
  no se pudo encontrar la función "checkDbType"_

3. The third one trying to look up which library the checkDbType function could be in.  As you can see we tried it installing DrugExposureDiagnostics but our attemp was unsuccesfull... the error was the same


I add some logs:

File log that is found in storage folder:
[BARDENA_log.txt](https://github.com/user-attachments/files/16080745/BARDENA_log.txt)

Trace of the attempts 1 and 2:
[Trace_Step_1.txt](https://github.com/user-attachments/files/16080842/Trace_Step_1.txt)
[Trace_Step_2.txt](https://github.com/user-attachments/files/16080843/Trace_Step_2.txt)

Attempt 3 image (it´s the image of @JavierGorri comment)
![Attempt3](https://github.com/oxford-pharmacoepi/MegaStudy/assets/131248056/fb61b1e7-9a85-4076-bc9a-81c117c3156b)

Thanks and regards

**Comment by tiozab** [comment]
Thanks @D229463 can you show me screenshot where the packages tab shows the "codelistgenerator" package please?

**Comment by D229463** [comment]
Sure, we have the 2.2.2 version

![image](https://github.com/oxford-pharmacoepi/MegaStudy/assets/131248056/1e1f07cc-b33c-425d-9fe5-504b28277a0a)

**Comment by tiozab** [comment]
great, because the checkDbType function comes from Codelistgenerator, can you show me the libraries you load in your codetorun script?

**Comment by D229463** [comment]
yea

![image](https://github.com/oxford-pharmacoepi/MegaStudy/assets/131248056/ed932363-4ca3-41f0-8769-b717d262f4a0)

**Comment by tiozab** [comment]
@D229463 the error does not make sense because checkDbType is a function from CodelistGenerator and no one has had this error before. https://darwin-eu.github.io/CodelistGenerator/articles/a04_Vocab_based_codelists.html can you try whether you can reproduce the vignette with your data?

**Comment by D229463** [comment]
@tiozab  ok, yestyerday we had run the script from the begginig with the space added (now it´s still running) and when it finishes I try to reproduce the vignette

**Comment by tiozab** [comment]
@D229463 ah I thought your code stopped at the drug cohort generation. That means this error was resolved?

**Comment by D229463** [comment]
No, we have tried it again (from the beginning avoiding re-executions)... but as I see in the log it is still in step "GENERATE INDICATION CONCEPTS" (last time it needed 13 hours in this step)

**Comment by D229463** [comment]
Bad news... I try the first step of the vignette and error is the same

![image](https://github.com/oxford-pharmacoepi/MegaStudy/assets/131248056/efc3d031-b935-42fb-af7d-68f9d00a0b40)

Do you want me to re-install the CodelistGenerator?

**Comment by tiozab** [comment]
@edward-burn can you have a look at this please?

**Comment by edward-burn** [comment]
@D229463 can I just confirm, if you clear your environment and restart r, does the following work 

cdm <- mockVocabRef()
getDrugIngredientCodes(cdm = cdm, name = "Adalimumab")

?

**Comment by D229463** [comment]
Now, I have tried to launch the vignette that tiozab told us before and the error has changed...Does it make sense?

![image](https://github.com/oxford-pharmacoepi/MegaStudy/assets/131248056/f8b60b9e-306b-4891-bbc8-d9ab7d628686)

**Comment by tiozab** [comment]
@D229463 Not really, but with everything anew, please try the DUS step once more with the 1.4 file and then we see where we get :-) we do not assess acetaminophen so maybe there is just s.th. wrong with that ingredient

**Comment by D229463** [comment]
We have run the script several times and it fails to create temporary tables on step "GENERATE DRUG CONCEPTS"

I add file logs:

1. File log that is found in storage folder: 
[BARDENA_log.txt](https://github.com/user-attachments/files/16260490/BARDENA_log.txt)

2. execution trace:
[script_log.txt](https://github.com/user-attachments/files/16260493/script_log.txt)

It has always failed in the "Created a temporary table" number 52. Log says "subindex out of bounds" but I don't know what index it is referring to

**Comment by tiozab** [comment]
@D229463 drug 52 is certolizumab. We have had problems with this drug in other places. I have modified the drugs.R script for certolizumab and cytarabine (the two drugs causing problems), can you replace this R script in your repo with this version and try again from "GENERATE DRUG CONCEPTS"
[drugs.zip](https://github.com/user-attachments/files/16262653/drugs.zip)

**Comment by D229463** [comment]
we have executed the new drugs.R file but we get the same error in the same "create table" :-(

![image](https://github.com/user-attachments/assets/e4f55100-b46e-4e7d-9221-38f4f931188f)

we have check that the error is in first step of drugs.R
[error_step_drugs.txt](https://github.com/user-attachments/files/16280913/error_step_drugs.txt)

**Comment by tiozab** [comment]
@edward-burn may it be a memory issue?

**Comment by JavierGorri** [comment]
@tiozab What can we do now? 
Thanks

**Comment by D229463** [comment]
Hi, are there some news about the error? 
thanks!

**Comment by tiozab** [comment]
@D229463 sorry I was on holiday.
@edward-burn need your help here!
